### PR TITLE
Clojars deployment toolchain

### PR DESCRIPTION
**Pull request by kelvinqian00**
Add Clojars deployment to our GitHub Workflows such that deployment happens upon each tag push.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/upload-artifact@v2
+      with:
+        name: colossal-squuid-artifact-${{ steps.version.outputs.VERSION }}
+        path: target/colossal-squuid-${{ steps.version.outputs.VERSION }}.jar
+
+    - name: Deploy to Clojars
+      run: make deploy VERSION=${{ steps.version.outputs.VERSION }}
+      env:
+        CLOJARS_USERNAME: ${{ secrets.CLOJARS_USERNAME }}
+        CLOJARS_PASSWORD: ${{ secrets.CLOJARS_PASSWORD }}
```
Comment (milt) [inline .github/workflows/deploy.yml:33]: Assuming that `CLOJARS_PASSWORD` here is a deploy token? https://clojars.org/tokens

Comment (kelvinqian00) [reply .github/workflows/deploy.yml:33]: Yes

Comment (milt) [reply .github/workflows/deploy.yml:33]: 👍

**Comment by kelvinqian00** [comment]
Huh GH lists the CD action despite occurring on a tag push.

**Review by milt — APPROVED** [review]
This looks good to me, build-clj is amazing.

**Comment by kelvinqian00** [comment]
Update to v0.1.2 as inspired by #13

**Pull request merged**